tests/storage/test_v2_instrument_history_readers.py

```
@@ -0,0 +1,1388 @@
+# write-capable H2c bootstrap migration vector.
+#
+# Before: only exchange_instruments exists (an already-accepted LKG row is
+# present); NO exchange_instrument_history, NO stage2_instrument_metadata_state.
+# After the write-capable bootstrap runs: both new tables exist, the revision
+# singleton is initialized from the resolved Stage2Config's own value, the
+# existing LKG row is completely untouched, NO speculative historical
+# backfill occurs (exchange_instrument_history stays empty until an operator
+# explicitly calls seed_current_instrument_history), and the feature path no
+# longer fails merely because the singleton was never initialized.
+# ============================================================================
+def test_legacy_pre_h2c_database_migration_vector():
```

Ruff (0.16.1)

[warning] Too many statements (51 > 50)

(PLR0915)